### PR TITLE
Add documentation about images and their lifecycles

### PR DESCRIPTION
**Pull request by michaelsauter**
The goal is to help admins and users understand which images are used,
and how the images and their deployments get updated.

Not perfect - but a start to document some answers to a reoccurring question.

FYI @borja44 @segator @maksymmaksym

### CODE REVIEW

File: docs/modules/administration/pages/image-lifecycle.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+1. Images used in the central ODS namespace
+2. Images used in the `*-cd` namespace of each ODS project
+3. Application images used in the `*-dev` and `*-test` namespace
+
+Every category has different lifecycles.
+
+=== Images used in the central ODS namespace
+
+Those images are owned by ODS admins. Examples of such images include the SonarQube image and the Nexus image. All of the images used by containers in the central ODS namespace are built by ODS during its installation. No container uses a vanilla image as provided by e.g.  DockerHub or quay.io.
```
Comment (felipecruz91) [inline docs/modules/administration/pages/image-lifecycle.adoc:17]: I'm not totally certain about what you want to convey with this message: ` No container uses a vanilla image as provided by e.g.  DockerHub or quay.io.` Do you mean that the images present in the `ods` namespace do not use any base image coming from a public registry like Dockerhub?

File: docs/modules/administration/pages/image-lifecycle.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Vulnerabilities in application images have a high impact, but mitigation is relatively easy. If the vulnerability is within the base image, and the latest image of the currently used image tag has the vulnerability fixed, one simply needs to trigger a new Jenkins job, which builds a new image, and deploys that new image. If the image tag referenced in the `FROM` instruction of the `Dockerfile` does not contain the security fix, one can attempt to change to an image tag which is not vulnerable. Finally, the vulnerability might not be in the base image, but in the `Dockerfile` of the application, in which the security fix can be applied there.
+
+Vulnerabilities in images used in the `*-cd` namespace of each ODS project are a bit harder to mitigate. This is because the vulnerability can be in the base image or in the ODS customisation of it. Further, to actually fix the vulnerability in the running container, one needs to built a new vulnerabilty-free image and have that rolled out to each `*-cd` namespace (remember there are no automatic image triggers).
```
Comment (felipecruz91) [inline docs/modules/administration/pages/image-lifecycle.adoc:47]: Nicely explained 👍

**Review by felipecruz91 — COMMENTED** [review]
This is a really good start!

File: docs/modules/administration/pages/image-lifecycle.adoc (hunk 1 above)
Comment (michaelsauter) [reply docs/modules/administration/pages/image-lifecycle.adoc:17]: Hmm yea - I meant that all deployments use images from local image streams. No deployment pulls directly from DockerHub.

**Review by segator — APPROVED** [review]
pretty clear, thanks!

**Pull request merged**